lib/src/modules/command.cc

```diff
@@ -0,0 +1,69 @@
+#include <pxp-agent/modules/command.hpp>
+#include <pxp-agent/configuration.hpp>
+#include <leatherman/file_util/file.hpp>
+#include <boost/filesystem/path.hpp>
+#include <boost/algorithm/string.hpp>
+
+namespace PXPAgent {
+namespace Modules {
+
+namespace fs = boost::filesystem;
+namespace lth_jc = leatherman::json_container;
+namespace lth_file = leatherman::file_util;
+
+static const std::string COMMAND_RUN_ACTION { "run" };
+
+Command::Command(const fs::path& exec_prefix, std::shared_ptr<ResultsStorage> storage) :
+    BoltModule { exec_prefix, std::move(storage) }
+{
+    module_name = "command";
+    actions.push_back(COMMAND_RUN_ACTION);
+
+    // Command actions require a single "command" string parameter
+    PCPClient::Schema input_schema { COMMAND_RUN_ACTION };
+    input_schema.addConstraint("command", PCPClient::TypeConstraint::String, true);
+    input_validator_.registerSchema(input_schema);
+
+    PCPClient::Schema output_schema { COMMAND_RUN_ACTION };
+    results_validator_.registerSchema(output_schema);
+}
+
+Util::CommandObject Command::buildCommandObject(const ActionRequest& request)
+{
+    const auto params = request.params();
+
+    assert(params.includes("command") &&
+           params.type("command") == lth_jc::DataType::String);
+
+    const fs::path& results_dir = request.resultsDir();
+
+    // Try to split the raw command text into executable and arguments, as
+    // leatherman::execution expects:
+    std::vector<std::string> command_parts;
+    auto raw_command = params.get<std::string>("command");
+
+    boost::split(command_parts,
```

Comment:
Will this function be affected by windows executables with spaces in their path? i.e. something like 
`C:\Program Files\OpenSSL\bin\openssl.exe`

Reply:
Additionally, do we want to fail early here if the executable does not include an extension on windows?

On windows the file extension is required to call any executable, will leatherman execute look for an extension? or just fail? I would assume we would just fail in that scenario since guessing file extensions seems fishy at best.

Reply:
It definitely won't handle that correctly - I'm not sure what a good heuristic is to split the user input up into executable and arguments in this case - would it be reasonable to expect the user to quote pathnames that have spaces in them within the command parameter, like `{"command": "\"C:\\Program Files\\OpenSSL\\bin\\openssl.exe\" ..."}`?

Reply:
Yes, if the path has spaces I believe it's reasonable to expect quotes.

Reply:
And yeah, it looks like leatherman execution is checking for file extensions [here](https://github.com/puppetlabs/leatherman/blob/master/execution/src/windows/execution.cc#L57)

Reply:
Agreed that quoting paths or arguments with spaces should be expected. https://stackoverflow.com/questions/21335041/how-to-split-a-string-using-shell-like-rules-in-c has some suggestions how, looks like https://www.boost.org/doc/libs/1_65_0/doc/html/boost/program_options/split_unix.html might cover it. Windows makes that trickier, but it might work well enough.

Would anything be easier if we made Bolt split the string?

Reply:
It turns out run_command is kind of weird, we're accepting something shell-specific but not running it in a shell. So we need to define what the characteristics are. Looks like bolt_shim went with whatever the semantics of https://www.rubydoc.info/stdlib/open3/Open3.capture3 are.